Modules/_io/stringio.c

```diff
@@ -839,8 +858,17 @@ stringio_getstate(stringio *self, PyObject *Py_UNUSED(ignored))
     return state;
 }
 
+/*[clinic input]
+@critical_section
+_io.StringIO.__setstate__
+
+    state: object
+
```

Comment:
```suggestion

    state: object
    /
```

Reply:
@colesbury 
Out of curiosity:
User should explicitly care about thread-safe when implementing [dunder methods of object](https://docs.python.org/3/library/pickle.html?highlight=__setstate__#object.__setstate__) for each under method?
Or Will it be handled from interpreter side?